zipline/pipeline/factors/factor.py

```diff
@@ -647,12 +647,14 @@ def zscore(self, mask=NotSpecified, groupby=NotSpecified):
         def zscore(row):
             return (row - nanmean(row)) / nanstd(row)
 
-        return GroupedRowTransform(
+        zscore_factor = GroupedRowTransform(
             transform=zscore,
             factor=self,
             mask=mask,
             groupby=groupby,
         )
+        zscore_factor.window_safe = True
+        return zscore_factor
```

Comment:
(In which case this should be passed as a keyword-argument to `GroupedRowTransform`.